yt_dlp/extractor/itv.py

```diff
@@ -105,28 +120,69 @@ def _real_extract(self, url):
                 href = ios_base_url + href
             ext = determine_ext(href)
             if ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
+                hls_formats, hls_subs = self._extract_m3u8_formats_and_subtitles(
                     href, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls', fatal=False))
+                    m3u8_id='hls', fatal=False)
+                formats.extend(hls_formats)
```

Comment:
Are you sure there is only one m3u8? otherwise, the `hls_subs` of everything but the last one is discarded